synapse/lib/storm.py

```diff
@@ -1414,3 +1415,65 @@ def _stormOperTask(self, query, oper):
         for tname in args:
             evt = ':'.join(['task', tname])
             [core.fire(evt, node=node, storm=True, **opts) for node in nodes]
+
+    def _stormOperTree(self, query, oper):
+
+        args = oper[1].get('args')
+        opts = dict(oper[1].get('kwlist'))
+
+        # Prevent bad recursions????
+        recurlim = opts.get('recurlim', 20)
+
+        srcp = None
+        dstp = args[0]
+
+        if len(args) > 1:
+            srcp = args[0]
+            dstp = args[1]
+
+        # do we have a relative source property?
+        relsrc = srcp is not None and srcp.startswith(':')
+
+        tufs = query.data()
+
+        queried_vals = set()
+
+        while True:
+
+            vals = set()
+
+            if srcp is not None and not relsrc:
+
+                for tufo in tufs:
+                    valu = tufo[1].get(srcp)
+                    if valu is not None:
+                        vals.add(valu)
+
+            elif not relsrc:
+
+                for tufo in tufs:
+                    form, valu = s_tufo.ndef(tufo)
+                    if valu is not None:
+                        vals.add(valu)
+
+            else:
+                for tufo in tufs:
+                    form, _ = s_tufo.ndef(tufo)
+                    valu = tufo[1].get(form + srcp)
+                    if valu is not None:
+                        vals.add(valu)
+
+            qvals = list(vals - queried_vals)
+            if not qvals:
+                break
+
+            [query.add(t) for t in self.stormTufosBy('in', dstp, qvals, limit=opts.get('limit'))]
+
+            queried_vals = queried_vals.union(vals)
+
+            if recurlim:
+                recurlim -= 1
+                if not recurlim:
```

Comment:
Might make more sense to replace this with `if recurlim < 1:`

Reply:
done; and added a type norm step to ensure we're working with a "int" value.